tests/unit/deploy/test_inferencer.py

```diff
@@ -0,0 +1,102 @@
+"""Tests for Torch and OpenVINO inferencers."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from collections.abc import Callable
+from pathlib import Path
+
+import pytest
+import torch
+
+from anomalib.data import MVTec, TaskType
+from anomalib.deploy import ExportMode, OpenVINOInferencer, TorchInferencer
+from anomalib.engine import Engine
+from anomalib.models import Padim
+from tests.legacy.helpers.inference import MockImageLoader
+
+
+@pytest.mark.parametrize(
+    "task",
+    [
+        TaskType.CLASSIFICATION,
+        TaskType.DETECTION,
+        TaskType.SEGMENTATION,
+    ],
+)
+def test_torch_inference(task: TaskType, ckpt_path: Callable[[str], Path], dataset_path: Path) -> None:
+    """Tests Torch inference.
+
+    Model is not trained as this checks that the inferencers are working.
+
+    Args:
+        task (TaskType): Task type
+        ckpt_path: Callable[[str], Path]: Path to trained PADIM model checkpoint.
+        dataset_path (Path): Path to dummy dataset.
+    """
+    model = Padim()
+    engine = Engine()
+    export_path = ckpt_path("Padim").parent.parent
```

Comment:
do we any plans to change this with the random models stuff in the future? Once we separate the image and video models based on different tasks.

Reply:
Yeah it would be nice to test this on random models. This way we can be sure that we can call infer on all the models.